cmd/kusk/internal/overlays/overlays.go

```diff
@@ -0,0 +1,185 @@
+package overlays
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"sigs.k8s.io/yaml"
+)
+
+const imageName = "jasmingacic/overlay-cli"
+
+type Overlay struct {
+	Overlays string   `json:"overlays,omitempty" yaml:"overlays,omitempty"`
+	Extends  string   `json:"extends,omitempty" yaml:"extends,omitempty"`
+	Actions  []Action `json:"actions,omitempty" yaml:"actions,omitempty"`
+	path     string
+	url      string
+}
+
+type Action struct {
+	Target string      `json:"target,omitempty" yaml:"target,omitempty"`
+	Remove bool        `json:"remove,omitempty" yaml:"remove,omitempty"`
+	Update interface{} `json:"update,omitempty" yaml:"update,omitempty"`
+	Where  interface{} `json:"where,omitempty" yaml:"where,omitempty"`
+}
+
+// type Update struct {
+// 	Title       string `json:"title,omitempty" yaml:"title,omitempty"`
+// 	Description string `json:"description,omitempty" yaml:"description,omitempty"`
+// }
+
+func NewOverlay(path string) (o *Overlay, err error) {
+	o = &Overlay{}
+	if !IsUrl(path) {
+		dat, err := os.ReadFile(path)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := yaml.UnmarshalStrict(dat, o); err != nil {
+			return nil, err
+		}
+		overlay, err := os.CreateTemp("", "overlay")
+		if err != nil {
+			return nil, err
+		} else {
+			if _, err := overlay.Write(dat); err != nil {
+				return nil, err
+			}
+		}
+		o.path = overlay.Name()
+
+		return o, nil
+	} else {
+		o, err = getFile(path)
+		if err != nil {
+			return nil, err
+		}
+		return o, nil
+	}
+}
+
+func (o *Overlay) Apply() (string, error) {
+	var err error
+	var overlayed string
+	if !IsUrl(o.Extends) {
+		overlayed, err = applyOverlay(o.path, o.Extends)
+		if err != nil {
+			return "", err
+		}
+	} else {
+		overlayed, err = applyOverlay(o.path, "")
+		if err != nil {
+			return "", err
+		}
+	}
+	f, err := os.CreateTemp("", "overlay")
+	if err != nil {
+		return "", err
+	} else {
```

Comment:
Can combine else { if...{...}} into just else if on one line

Reply:
changed it according to your suggestion. Thanks